docs/guides/volumesnapshot/overview/index.md

```diff
@@ -12,73 +12,90 @@ menu_name: docs_{{ .version }}
 section_menu_id: guides
 ---
 
-# VolumeSnapshot Using Stash
+# VolumeSnapshot Using KubeStash
 
-This guide will give you an overview of how VolumeSnapshot process works in Stash.
+This guide will give you an overview of how VolumeSnapshot process works in KubeStash.
 
 ## How Backup Process Works?
 
-The following diagram shows how Stash creates VolumeSnapshot via Kubernetes native API. Open the image in a new tab to see the enlarged version.
+The following diagram shows how KubeStash creates VolumeSnapshot via Kubernetes native API. Open the image in a new tab to see the enlarged version.
 
 <figure align="center">
-  <img alt="Stash Backup Flow" src="/docs/guides/volumesnapshot/overview/images/volumesnapshot-overview.svg">
-<figcaption align="center">Fig: Volume Snapshotting Process in Stash</figcaption>
+  <img alt="KubeStash Backup Flow" src="/docs/guides/volumesnapshot/overview/images/volumesnapshot-overview.svg">
+<figcaption align="center">Fig: Volume Snapshotting Process in KubeStash</figcaption>
 </figure>
 
 The `VolumeSnapshot` process consists of the following steps:
 
-1. At first, a user creates a `BackupConfiguration` crd which specifies the targeted workload or targeted PVC.
+1. At first, a user creates a `Secret` with access credentials of the backend where the backed-up data will be stored. 
 
-2. Stash operator watches for `BackupConfiguration` crd.
+2. Then, she creates a `BackupStorage` custom resource that specifies the backend information, along with the `Secret` containing the credentials needed to access the backend. 
 
-3. When it finds a `BackupConfiguration` crd, it creates a `CronJob` to take a periodic backup of the target volumes.
+3. KubeStash operator watches for `BackupStorage` custom resources. When it finds a `BackupStorage` object, it initializes the `BackupStorage` by uploading the `metadata.yaml` file into the target storage. 
 
-4. The `CronJob` triggers backup on each scheduled time slot by creating a `BackupSession` crd.
+4. Then, she creates a `BackupConfiguration` custom resource, targeting either a workload application or a standalone volume. The `BackupConfiguration` object specifies the `Repository` pointing to a `BackupStorage` that contains backend information, indicating where to upload backup data. It also defines the `Addon` information with a specified task and their configuration parameters to be used for backing up the volumes. 
```

Comment:
```suggestion
4. Then, she creates a `BackupConfiguration` custom resource, targeting either a workload application or a standalone volume. The `BackupConfiguration` object specifies the `Repository` pointing to a `BackupStorage` that contains backend information, indicating where to upload backup data. It also defines the `Addon` information with a specified task and their configuration parameters to be used for backing up the volumes. 
```
Then, she creates a `BackupConfiguration` custom resource that specifies the targeted workload/PVC, the Addon info with specified tasks, etc. It also provides information about one or more repositories, each indicating a path and a `BackupStorage` for storing the metadata.